src/CredentialAgent.js

```diff
@@ -0,0 +1,59 @@
+import fetch from './fetch';
+import { Login } from '.';
+
+/**
+ * A credential agent that fetches the credentials from the taskcluster
+ * login service, given an access_token.
+ *
+ * Users can keep a single such credential agent and use it simultaneously
+ * with many Client instances.
+ */
+export class OIDCCredentialAgent {
+  constructor({accessToken, oidcProvider, loginBaseUrl}) {
+    this._accessToken = accessToken;
+    this.oidcProvider = oidcProvider;
+  }
+
+  // Update the access token, invalidating any cached credentials
+  set accessToken(accessToken) {
+    this._accessToken = accessToken;
+    delete this.credentialsPromise;
+  }
+
+  getCredentials(client) {
+    if (this.credentialsPromise && this.credentialsExpire > new Date()) {
+      return this.credentialsPromise;
+    }
+    
+    // Call the oidcCredentials endpoint with the access token.
+    const login = new Login();
+    const url = login.buildUrl(login.oidcCredentials, this.oidcProvider);
+    this.credentialsPromise = fetch(url, {
```

Comment:
Insert a blank line between the declaration and other statements, and put each method chain on its own indentation:

```js
this.credentialsPromise = fetch(url, {
  header: {
    Authorization: `Bearer ${this._accessToken}`
  }
})
.then(response => /* ... */)
.catch(err => /* ... */);


```